nautobot/extras/models/groups.py

```diff
@@ -1008,19 +850,25 @@ def get_siblings(self, include_self=False):
 
         return siblings.exclude(pk=self.pk)
 
+    # TODO: this is an interesting definition of "root node", as a node with no children has is_root() = False??
+    # TODO: unused in core
     def is_root(self):
         """Return whether this is a root node (has children, but no parents)."""
         return self.children.exists() and not self.parents.exists()
 
+    # TODO: this is an interesting definition of "leaf node", as a node with no parents has is_leaf() = False??
```

Comment:
Likewise, this should probably just be `type != "dynamic-set" and not self.children.exists()`

Reply:
Actually, `type != "dynamic-set"` should be sufficient

Reply:
Agreed, but also not touching this for now.